File: src/org/newtonpolyhedron/entity/vector/VectorFormat.java
package org.newtonpolyhedron.entity.vector;


public interface VectorFormat<Component extends Comparable <Component>,Vector extends AbstractVector <Component, Vector>> {
	
	public Component[] createArrayOfZeros(int length);
	
	public Component parseElement(String src);
	
	public Vector makeVector(Component[] components);
}

File: src/org/newtonpolyhedron/entity/vector/VectorFormat.scala
package org.newtonpolyhedron.entity.vector

trait VectorFormat[C <: Ordered[C], V <: MathVector[C, V]] {
  def createArrayOfZeros(length: Int): Array[C]

  def parseElement(src: String): C

  def makeVector(components: Seq[C]): V
}